reference-implementation/lib/transform-stream.js

```diff
@@ -363,14 +386,18 @@ module.exports = class TransformStream {
 
     this._readable = new ReadableStream(source, transformer.readableStrategy);
 
-    const sink = new TransformStreamSink(this, startPromise);
+    const sink = new TransformStreamSink(this, startPromise.then(() => {
+      return TransformStreamBackpressureGonePromise(this);
+    }));
```

Comment:
I don't see the point of doing this in the constructor instead of inside sink.start. It's conceptually something that .start does, and it's symmetrical with what source.start does. It's not really any clearer this way.